src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoaderCursor.cs

```diff
@@ -225,12 +225,64 @@ private Delegate CreateGetterDelegate<TValue>(int col)
                 {
                     getterDelegate = CreateUInt64GetterDelegate(colInfo);
                 }
+                else if (typeof(TValue) == typeof(VBuffer<bool>))
+                {
+                    getterDelegate = CreateVBufferBooleanGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<byte>))
+                {
+                    getterDelegate = CreateVBufferByteGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<DateTime>))
+                {
+                    getterDelegate = CreateVBufferDateTimeGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<double>))
+                {
+                    getterDelegate = CreateVBufferDoubleGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<short>))
+                {
+                    getterDelegate = CreateVBufferInt16GetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<int>))
+                {
+                    getterDelegate = CreateVBufferInt32GetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<long>))
+                {
+                    getterDelegate = CreateVBufferInt64GetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<sbyte>))
+                {
+                    getterDelegate = CreateVBufferSByteGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<float>))
+                {
+                    getterDelegate = CreateVBufferSingleGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<ReadOnlyMemory<char>>))
+                {
+                    getterDelegate = CreateVBufferStringGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<ushort>))
+                {
+                    getterDelegate = CreateVBufferUInt16GetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<uint>))
+                {
+                    getterDelegate = CreateVBufferUInt32GetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(VBuffer<ulong>))
+                {
+                    getterDelegate = CreateVBufferUInt64GetterDelegate(colInfo);
+                }
                 else
                 {
                     throw new NotSupportedException();
                 }
 
-                return getterDelegate as ValueGetter<TValue>;
+                return getterDelegate;
```

Comment:
There is some behavior change here. Doing this cast here will check if the Delegate is of the wrong type. If it is, this method will return `null`, which will trigger an assert while creating the getters, but then store `null` in the delegate array. If you return a delegate of the wrong type now, the assert isn't triggered, and an exception will occur when someone tries using the Delegate out of the array.